src/status_im/contexts/onboarding/events.cljs

```diff
@@ -165,23 +346,35 @@
  :onboarding/finalize-setup
  (fn [{db :db}]
    (let [{:keys [password syncing? auth-method
-                 temporary-display-name?]} (:onboarding/profile db)
-         {:keys [key-uid] :as profile}     (:profile/profile db)
-         biometric-enabled?                (= auth-method constants/auth-method-biometric)]
+                 temporary-display-name?
+                 enable-news-notifications?
+                 enable-notifications?]} (:onboarding/profile db)
+         {:keys [key-uid] :as profile}   (:profile/profile db)
+         biometric-enabled?              (= auth-method constants/auth-method-biometric)]
      {:db (assoc db :onboarding/generated-keys? true)
-      :fx [(when temporary-display-name?
-             [:dispatch [:profile/set-default-profile-name profile]])
-           (when biometric-enabled?
-             [:keychain/save-password-and-auth-method
-              {:key-uid         key-uid
-               :masked-password (if syncing?
-                                  password
-                                  (security/hash-masked-password password))
-               :on-success      (fn []
-                                  (rf/dispatch [:onboarding/set-auth-method auth-method])
-                                  (when syncing?
-                                    (rf/dispatch
-                                     [:onboarding/finish-onboarding false])))
-               :on-error        #(log/error "failed to save biometrics"
-                                            {:key-uid key-uid
-                                             :error   %})}])]})))
+      :fx (cond->
+            [(when temporary-display-name?
+               [:dispatch [:profile/set-default-profile-name profile]])
+             (when biometric-enabled?
+               [:keychain/save-password-and-auth-method
+                {:key-uid         key-uid
+                 :masked-password (if syncing?
+                                    password
+                                    (security/hash-masked-password password))
+                 :on-success      (fn []
+                                    (rf/dispatch [:onboarding/set-auth-method auth-method])
+                                    (when syncing?
+                                      (rf/dispatch
+                                       [:onboarding/finish-onboarding])))
+                 :on-error        #(log/error "failed to save biometrics"
+                                              {:key-uid key-uid
+                                               :error   %})}])]
+
+            (and enable-notifications?
+                 enable-news-notifications?)
+            (conj [:dispatch
+                   [:notifications/news-notifications-switch true]])
+
+            enable-notifications?
+            (conj [:dispatch
+                   [:push-notifications/switch true]]))})))
```

Comment:
Here we're expanding the `:onboarding/finalize-setup` event to handle the activation of the push notification settings if the user has opted-in.